src/controls/PrimitiveArrayControl.tsx

```diff
@@ -55,9 +55,13 @@ export function PrimitiveArrayControl({
   )
 
   const prevDataValue = usePreviousValue(data)
+  const minItems = rootSchema.properties?.[path].minItems || 0
+  const length = minItems === 0 ? 1 : minItems
   useEffect(() => {
     if (data === undefined && prevDataValue === null) {
-      addDefaultItemToList()
+      for (let i = 0; i < length; i++) {
+        addDefaultItemToList()
```

Comment:
Why is this necessary? I'm a bit worried this will cause a bunch of state updates in cases where it's used

Reply:
Instead of looping here, you could just change the initial value from [undefined] to range(minItems).map(()=> undefined) and I thibk that'd read better

Reply:
@DrewHoo I'm not following how that would solve the problem I describe above. At best I am able to get an initial start state of list: [0, null, null]

Reply:
I see what you mean. Looking this, I don't think the current default value behavior makes sense. The jsonforms implementation of createDefaultValue just returns 0 if the schema type is numeric. How do you feel about using `undefined` except in cases where a `default` property is set?